docs/ir/ir-breaking-changes.rst

```diff
@@ -116,6 +116,44 @@ Previously `f()` would return `0x64656164626565663135646561640000000000000000000
 Now it is returning `0x6465616462656566000000000000000000000000000000000000000000000010` (it has correct length, and correct elements, but doesn't contain dirty data).
 
 
+* For the old code generator, the evaluation order is unspecified. For the new code generator, we try to evaluate in source order (left to right), but do not guarantee it. This can lead to semantic differences.
+
+For example:
+
+::
+     // SPDX-License-Identifier: GPL-3.0
+     pragma solidity >0.8.0;
+     contract C {
+         function preincr_u8(uint8 a) public pure returns (uint8) {
+             return ++a + a;
+         }
+     }
+
+The function `preincr_u8(uint8)` returns the following for the input 1:
+- Old code gen: 3 (`1 + 2`) but the return value is unspecified in general
+- New code gen: 4 (`2 + 2`) but the return value is not guaranteed
+
+This behavior also true for function argument expressions.
+
+For example:
+
+::
+     // SPDX-License-Identifier: GPL-3.0
+     pragma solidity >0.8.0;
+     contract C {
+         function identity(uint8 a) public pure returns (uint8) {
+             return a;
+         }
+         function g(uint8 b) public pure returns (uint8) {
+             return identity(++b + b);
```

Comment:
Isn't it supposed to show the order in which separate parameters are evaluated? I think it does not show that with just one argument. I think an example like this would be more suitable here:

```suggestion
             return identity(++a + ++b, ++a + ++b);
```

Reply:
> I think it does not show that with just one argument

Why not? `++b + b` will be different depending on left-to-right evaluation and vice versa. If `b` is initially `1`, the left-to-right gives us `2 + 2`, `1 + 2` otherwise (right-to-left).

Reply:
Yes it will be different but this example seems redundant to me. The one above already shows the evaluation order in expressions and I don't see why it would be different when the expression is placed in a function call.

So the only way including this example makes sense to me is if you wanted to show evaluation order between expressions included in different arguments.

Reply:
> Yes it will be different but this example seems redundant to me. The one above already shows the evaluation order in expressions and I don't see why it would be different when the expression is placed in a function call.

Ah, now I see what you mean. Will update.

Reply:
Updated. Good catch! Looks like function argument evaluation order actually does not differ across old and new code gens.